lib/match.js

```diff
@@ -11,45 +12,21 @@ const meta = {
   },
 };
 
-const aliases = {
-  pascalcase: /^[A-Z]([A-Z0-9]*[a-z]+)+[A-Z0-9]*(?:\..*)?$/,
-  camelcase: /^[a-z]+((\d)|([A-Z0-9][a-z0-9]+))*([A-Z])?(?:\..*)?$/,
-  snakecase: /^([a-z]+_)*[a-z]+(?:\..*)?$/,
-  kebabcase: /^([a-z]+-)*[a-z]+(?:\..*)?$/,
-};
-
-aliases.PascalCase = aliases.pascalcase;
-aliases.camelCase = aliases.camelcase;
-aliases.snake_case = aliases.snakecase;
-aliases['kebab-case'] = aliases.kebabcase;
-
-const getRegex = (value, filename) => {
-  if (value instanceof RegExp) return [value, value.toString()];
-  if (typeof value === 'string') {
-    const regex = aliases[value];
-    if (!regex) throw new Error(`Unrecognized option "${value}"`);
-    return [regex, value];
-  }
-
-  const extension = filename.substr(filename.lastIndexOf('.'));
-  const valueForExtension = value[extension];
-  return valueForExtension ? getRegex(valueForExtension) : [];
-};
-
 module.exports = {
   meta,
   create: (context) => ({
     Program: (node) => {
       const filename = context.getFilename();
-      const basename = path.basename(filename);
-      const [regex, regexStr] = getRegex(context.options[0], basename);
+      const includePath = !!context.options[0]?.includePath;
```

Comment:
I don't think this project is set up to support optional prop chaining

```
SyntaxError: /Users/mark/code/eslint-plugin-filename-rules/lib/match.js: Unexpected token (20:47)
  18 |     Program: (node) => {
  19 |       const filename = context.getFilename();
> 20 |       const includePath = !!context.options[0]?.includePath;
     |                                                ^
  21 |       const name = includePath ? filename : path.basename(filename);
  22 |       const [regex, regexStr] = getRegex(context.options[0], name);
  23 |       if (!regex) return;
    at Parser.pp$5.raise (/Users/mark/code/eslint-plugin-filename-rules/node_modules/babylon/lib/index.js:4454:13)
```

Reply:
@TSMMark this project doesnt do any transpilation, so any syntax is only supported if your Node supports it. Are you using Node < v14? I can update package.json with a node version requirement. It's probably not worth supporting Node versions below 14, especially in the context of a development environment.